ouroboros-network/src/Ouroboros/Network/Mux/Egress.hs

```diff
@@ -138,10 +138,12 @@ mux :: (MonadSTM m)
      => TVar m Int
      -> PerMuxSharedState ptcl m
      -> m ()
-mux cnt pmss = do
+mux cnt pmss = go
+  where
+  go = do
     w <- atomically $ readTBQueue $ tsrQueue pmss
     case w of
-         TLSRDemand mid md d -> processSingleWanton pmss mid md d cnt >> mux cnt pmss
+         TLSRDemand mid md d -> processSingleWanton pmss mid md d cnt >> go
```

Comment:
If you are this, maybe rename `d` as `wanton`; this will agree with the name in other places which is used for this variable.